internal/observability/event/node_formatter_audit_jsonx_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package event
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/logical"
+
+	"github.com/hashicorp/vault/sdk/helper/jsonutil"
+
+	"github.com/hashicorp/eventlogger"
+
+	"github.com/stretchr/testify/require"
+)
+
+// fakeJSONXAuditEvent will return a new fake event containing audit data based
+// on the specified auditSubtype and logical.LogInput.
+func fakeJSONXAuditEvent(t *testing.T, subtype auditSubtype, input *logical.LogInput) *eventlogger.Event {
+	t.Helper()
+
+	date := time.Date(2023, time.July, 11, 15, 49, 10, 0, time.Local)
+
+	auditEvent, err := newAudit(
+		WithID("123"),
+		WithSubtype(string(subtype)),
+		WithFormat(string(AuditFormatJSONX)),
+		WithNow(date),
+	)
+	require.NoError(t, err)
+	require.NotNil(t, auditEvent)
+	require.Equal(t, "123", auditEvent.ID)
+	require.Equal(t, "v0.1", auditEvent.Version)
+	require.Equal(t, AuditFormatJSONX, auditEvent.RequiredFormat)
+	require.Equal(t, subtype, auditEvent.Subtype)
+	require.Equal(t, date, auditEvent.Timestamp)
+
+	auditEvent.Data = input
+
+	e := &eventlogger.Event{
+		Type:      eventlogger.EventType(AuditType),
+		CreatedAt: auditEvent.Timestamp,
+		Formatted: make(map[string][]byte),
+		Payload:   auditEvent,
+	}
+
+	return e
+}
+
+// TestNewAuditFormatterJSONX ensures we can create new AuditFormatterJSONX structs.
+func TestNewAuditFormatterJSONX(t *testing.T) {
+	f := NewAuditFormatterJSONX()
+	require.NotNil(t, f)
+}
+
+// TestAuditFormatterJSONX_Reopen ensures that we do no get an error when calling Reopen.
+func TestAuditFormatterJSONX_Reopen(t *testing.T) {
+	require.NoError(t, NewAuditFormatterJSONX().Reopen())
+}
+
+// TestAuditFormatterJSONX_Type ensures that the node is a 'formatter' type.
+func TestAuditFormatterJSONX_Type(t *testing.T) {
+	require.Equal(t, eventlogger.NodeTypeFormatter, NewAuditFormatterJSONX().Type())
+}
+
+// TestAuditFormatterJSONX_Process attempts to run the Process method to convert
+// pre-formatted JSON to XML (JSONx).
+func TestAuditFormatterJSONX_Process(t *testing.T) {
+	tests := map[string]struct {
+		IsErrorExpected      bool
+		ExpectedErrorMessage string
+		Subtype              auditSubtype
+		Data                 *logical.LogInput
+	}{
+		"request-no-formatted-json": {
+			IsErrorExpected:      true,
+			ExpectedErrorMessage: "event.(AuditFormatterJSONX).Process: pre-formatted JSON required but not found: invalid parameter",
+			Subtype:              AuditRequest,
+			Data:                 nil,
+		},
+		"response-no-formatted-json": {
+			IsErrorExpected:      true,
+			ExpectedErrorMessage: "event.(AuditFormatterJSONX).Process: pre-formatted JSON required but not found: invalid parameter",
+			Subtype:              AuditResponse,
+			Data:                 nil,
+		},
+		"request-basic-json": {
+			IsErrorExpected: false,
+			Subtype:         AuditRequest,
+			Data:            &logical.LogInput{Type: "magic"},
```

Comment:
it'd be interesting to add a test case for a complex json with some field nesting.